jmix-superset/superset/src/main/java/io/jmix/superset/schedule/SupersetTokenScheduleConfigurer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.superset.schedule;
+
+import io.jmix.superset.SupersetProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Component;
+
+@Component("sprset_SupersetTokenScheduleConfigurer")
+public class SupersetTokenScheduleConfigurer {
+    private static final Logger log = LoggerFactory.getLogger(SupersetTokenScheduleConfigurer.class);
+
+    private final TaskScheduler taskScheduler;
+    private final TaskScheduler csrfTaskScheduler;
+    private final SupersetProperties supersetProperties;
+    private final SupersetTokenManager accessTokenManager;
+
+    private boolean csrfTokenSchedulerStarted = false;
+
+    public SupersetTokenScheduleConfigurer(@Qualifier("sprset_ThreadPoolAccessTokenTaskScheduler")
+                                           TaskScheduler taskScheduler,
+                                           @Autowired(required = false)
+                                           @Qualifier("sprset_ThreadPoolCsrfTokenTaskScheduler")
+                                           TaskScheduler csrfTaskScheduler,
+                                           SupersetProperties supersetProperties,
+                                           SupersetTokenManager accessTokenManager) {
+        this.taskScheduler = taskScheduler;
+        this.csrfTaskScheduler = csrfTaskScheduler;
+        this.supersetProperties = supersetProperties;
+        this.accessTokenManager = accessTokenManager;
+    }
+
+    @EventListener
+    public void onContextRefreshedEvent(ContextRefreshedEvent event) {
+        startAccessTokenScheduler();
```

Comment:
What do you think about adding the configuration property that will enable/disable automatic tokens refresh? Something like `jmix.superset.tokens-refresh-enabled=true`. True be default, if set to false, then tasks for tokens refresh won't be created. This will allow users to manage tokens refreshing  themself (use Quartz, custom refreshing intervals, etc.)

Reply:
Fixed